### DISCUSSION TITLE
Nilan Compact P support

### DISCUSSION THREAD

**Discussion by Pepekliste** [veista/nilan#113] [Q&A] [answered]
Hello, I own a Nilan Compact unit and I bought a Waveshare RS485 to ETH converter for it. Unfortunately, I can't get it connected and working... I have the unit connected to the converter with three wires, A B and GND. The converter flashes blue when connected. However, I am unable to set up and connect to Home Assistant. I have also tried getting data through NodeRed, however this has restarted the unit several times so I would hate to repeat, the result was equally unsuccessful.. I am not IT and don't know how to deal with this. @veista Are 120Ohm resistors a necessity? And is Compact supported?

**Comment by veista** [comment]
You need to provide a bit more information:

1. Picture of device type plate
2. Picture of HMI as described in read me
3. Is your bridge the type B? The type A Does not work

**Comment by Pepekliste** [comment]
![IMG_0602](https://github.com/veista/nilan/assets/113250773/e48fede4-79c5-45b5-85a1-fcbeacc0f300)
![IMG_0600](https://github.com/veista/nilan/assets/113250773/18b361da-e393-42dd-9da4-09dfb39c5aaa)
![IMG_0599](https://github.com/veista/nilan/assets/113250773/1c38b26e-5058-424b-9e0a-ceabb30eee28)
![IMG_0598](https://github.com/veista/nilan/assets/113250773/ef533246-d7e5-410b-b968-3e0a39e435bd)
![IMG_0597](https://github.com/veista/nilan/assets/113250773/f04cf906-8094-4e71-bf61-93995c0e1617)
![IMG_0596](https://github.com/veista/nilan/assets/113250773/9a331608-a4b3-4e23-8882-6797146a347f)
![IMG_0595](https://github.com/veista/nilan/assets/113250773/a8c34530-8216-4c1b-9397-52593fc916fd)

Sending some photos, hope it will help

**Reply by veista** [reply]
Well from the pictures your com connector on the waveshare bridge is not pushed in all the way.

It should work, but your model is not supported yet.

**Reply by veista** [reply]
Also you should use a single twisted pair on A and B. Now you have one pair on A and one pair on B.

**Reply by veista** [reply]
https://github.com/veista/nilan/wiki/Wiring-for-CTS602

**Reply by veista** [reply]
It seems to me that your whole connector is flipped. 1 -> 8 and so on.

**Reply by Pepekliste** [reply]
Sorry, it took me a while to get back to it. This is how my supplier wired it... I fixed it according to your instructions and I'm going to try. I'll be in touch.

**Reply by veista** [reply]
Seems to me your supplier is terrible at his job :/

**Reply by Pepekliste** [reply]
It looks like it, but I'm glad that after your advice the unit is already communicating and that it's not a fault in the unit, just bad wiring..

**Reply by veista** [reply]
Does your device have any other type plate? Do you have a manual for this device?

**Reply by Pepekliste** [reply]
What do You need, I can ask my delivering company. I had some manual, but I cant find it. What do You need? Modbus registers?

**Reply by veista** [reply]
I need which registers are actually used by the device. I usually figure this out from the manual. I get the manual from nilan website. Cant find anything with your picture.

Nilan has regional trade names of devices. Type plate usually has the global info. The picture you gave is not the usual type plate.

**Reply by Pepekliste** [reply]
![IMG_0811](https://github.com/veista/nilan/assets/113250773/0d97d02e-3a38-4324-8462-f2fa0c6ae56d)
Is this better?

**Reply by veista** [reply]
No, but its one more puzzle piece :)

**Reply by Pepekliste** [reply]
I sent a request to the supplier. I hope they respond soon.

**Reply by veista** [reply]
Ok. All starts to be making sense. Does your model have Air/Geo support?

**Reply by veista** [reply]
In other words do you have an outdoor external unit connected or ground source heating?

**Reply by Pepekliste** [reply]
No I don’t.. Just the unit itself.

**Reply by veista** [reply]
Ok, That explains a lot. Should be pretty straight forward to add. Will release it today!

**Reply by Pepekliste** [reply]
Super news!

**Comment by Pepekliste** [comment]
Its disconnected on purpose for now. Do You think its possible to have this model supported in near future? Or do I need to purchase official gateway and resign myself to not having the unit integrated into the HA?

**Reply by veista** [reply]
It should be possible to add once you get the comms. Working and follow the instructions in read me.

**Comment by Pepekliste** [comment]
I don’t know what to look for. Even if I correctly enabled the logging.. I’m sending some log, will it help? Or how should I enable logging for this integration? 
[home-assistant.log](https://github.com/veista/nilan/files/14140322/home-assistant.log)

**Reply by veista** [reply]
When you try to install, the integration will tell you: unsupported device. This means the comms are working. At this point if you have debug logging on, it would print your device info in the logs. I need that info.

**Reply by veista** [reply]
You should filter the logs and look for custom.component.nilan

**Reply by veista** [reply]
https://www.home-assistant.io/integrations/logger/

**Reply by Pepekliste** [reply]
[home-assistant 2.log](https://github.com/veista/nilan/files/14140508/home-assistant.2.log)
Type 44 thats it?

**Reply by veista** [reply]
Yes.

**Comment by veista** [comment, marked as answer]
I added support for your device in the latest beta. Be sure to have betas enabled in HACS if you are using HACS.

Please do the following: 
1. Report your modbus version shown by the integration
2. Report any unresponsive/non-working entities
3. Cross-check all functionality with the HMI panel

**Reply by Pepekliste** [reply]
Yes, it´s connected finaly! 
There are only two entities available, is it right? I have nothing to check.
![IMG_0160](https://github.com/veista/nilan/assets/113250773/15b2689d-3f30-4778-a057-f006a496dcc9)

**Reply by veista** [reply]
You should refresh the web page

**Reply by Pepekliste** [reply]
![IMG_0161](https://github.com/veista/nilan/assets/113250773/5e7875c0-9370-484c-ab92-97240fd2a981)
Going to check funcionality.

**Reply by Pepekliste** [reply]
![IMG_0163](https://github.com/veista/nilan/assets/113250773/61b1668a-0b4e-4da2-9f89-d1bfb797ed3f)
I haven't gone through everything yet, but it seems to work as it should! Many thanks for your time, let me buy you a coffee!

**Reply by Pepekliste** [reply]
https://www.buymeacoffee.com/veista/c/8367218

**Reply by veista** [reply]
Thank you for your support!

**Comment by bboto32** [comment]
Hi Pepakliste,

I will have my Nilan Compact delivered in two weeks and I would like to connect it right away. I feel like I am missing something, but I have not found any (step-by-step) instructions anywhere and don't know where to start, what to buy, where to place it, how to wire it... Could you please either send me the instructions you used to make it work or would you be so nice and write down step-by-step instructions how you managed it? I guess it would help also others in the future. Thank you!

**Reply by Pepekliste** [reply]
@bboto32 Is it still relevant? Theoretically, I could get to it over the Christmas holidays. If I remember everything, it's been a while.

**Reply by bboto32** [reply]
Not anymore, but thanks. I managed to successfully connect it. I spent several hours on it, with a guideline it would be quite easy. I wonder, why it doesn't exists here already. I will write it down and share to others.

**Comment by sandi-jersic** [comment]
Hello, I am having the same problem here, trying to establish connection to HA over ethernet using Waveshare RS485 to ETH adapter. I managed to set it up, wired it up like it says in design but all I see in Waveshare admin is HA attempts trying to communicate with Nilan with no response. What am I missing here? ChatGPT said that Nilan might use its internal communication protocol over RS485 and that is why it does not respond to Waveshare, but why some of you guys managed to work it out? I already spent a half a day digging into possible reasons but no luck.
  
Any help would be appreciated, thanks in advance!

Device: Nilan Compact P with mobile app cloud control

<img width="869" height="728" alt="waveshare transmission" src="https://github.com/user-attachments/assets/49983de7-4f0f-49e3-a92d-d3f47df520b4" />
<img width="869" height="728" alt="waveshare settings" src="https://github.com/user-attachments/assets/c72f12d0-b09a-47b3-99c7-de48ea802ba6" />
<img width="869" height="728" alt="waveshare settings" src="https://github.com/user-attachments/assets/05e64e19-aee8-4f36-b082-583cb0f4e8eb" />
<img width="869" height="728" alt="waveshare settings" src="https://github.com/user-attachments/assets/ca25034f-2cbe-4503-9d35-71d247c13119" />
<img width="869" height="728" alt="waveshare settings" src="https://github.com/user-attachments/assets/2d59e833-55f2-4049-952f-d6cab10da6a4" />
<img width="869" height="728" alt="waveshare settings" src="https://github.com/user-attachments/assets/def9b3d8-075c-496e-872b-f09ef34fa5c8" />

**Reply by veista** [reply]
I am not sure, but I believe you have the wrong adapter.

**Reply by sandi-jersic** [reply]
I read your manuals again and bought the right adapter and it works now! Thanks a lot!

**Reply by sandi-jersic** [reply]
I managed to build a view in Home assistant like the one on the display. Still needs some fix but good for a start:

<img width="585" height="400" alt="Screenshot 2026-02-12 at 12 53 43" src="https://github.com/user-attachments/assets/ed907061-222b-415d-bc8b-1e3b7e547497" />

**Reply by bboto32** [reply]
Nice, can you share how you made it please?

**Reply by sandi-jersic** [reply]
I took a picture of the display of my unit, put it in chatGPT which returned me that "house". Then I made configuration in HA for dashboard card: "Picture elements card configuration". The complete yaml is in attachment. Click on temperatures or ventilation opens popup with more settings or just graphs.

[nilan_dashboard_config.yaml](https://github.com/user-attachments/files/25368300/nilan_dashboard_config.yaml)
